File: greedy-algorithms/02-big-number-logical-solution.js
const readline = require('readline');
const rl = readline.createInterface({ 
	input: process.stdin,
	output: process.stdout 
}); 
let input = [];
rl.on('line', function (line) { 
	input.push(line.split('').map((el) => parseInt(el))) 
    
}) 
.on('close', function () {
    input[1].sort() // 입력이 다 되고 나서, close할 때 확인하자.
    let first = input[1][input[0][0]-1]
    let second = input[1][input[0][0]-2]

    let count = parseInt(input[0][1] / (input[0][2]+1)) * input[0][2]
    count += input[0][1] % (input[0][2] + 1)
    
    let result = 0
    result += (count) * first
    result += (input[0][1] - count) * second
	console.log(result); 
	process.exit();
}); 

File: python/greedy-algorithms/02-big-number-logical-solution.py
n, m, k = map(int, input().split())  # 예제로부터 주어진 첫번째 라인 입력값 받기

data = list(map(int, input().split()))  # 2번째 입력 값을 배열로 받아오기

data.sort()  # 작은 수 ~ 큰 수로 정렬하기 / 여기서 중복되는 값은 제거하지 않았다.

first = data[n-1]  # 첫번째로 큰 수
second = data[n-2]  # 두번째로 큰 수


# int(m/(k+1)) * k + m % (k + 1)
# 수열의 패턴을 인지해서 반복적으로 더하는 과정을 이해해야한다.

count = int(m/(k+1)) * k
count += m % (k+1)

result = 0
result += count * first
result += (m-count) * second

print(result)
